src/store/DiodeStore.js

```diff
@@ -115,10 +122,14 @@ class DiodeStore {
       const innerFragmentKeys = Object.keys(innerFragment);
 
       if (innerFragmentKeys.length === 0) {
-        // fetch all
-        if (cachedFragment && !cachedFragment[FETCH_ALL_CACHE]) {
-          filteredFragments[fragmentKey] = {};
+        // If fragment is already cached and all data in the fragment is
+        // already fetched, remove all fragments. Otherwise, fetch all
+        // fragments.
+        if (cachedFragment && cachedFragment[FETCH_ALL_CACHE]) {
+          return;
```

Comment:
If `innerFragmentKeys.length === 0` and `cachedFragment` is not fetch-all, we will need to fetch it anyway because we don't know if the cached fragment has all the data or not.